src/Foundatio/Storage/FolderFileStorage.cs

```diff
@@ -77,22 +81,32 @@ public class FolderFileStorage : IFileStorage {
             path = path.NormalizePath();
 
             string file = Path.Combine(Folder, path);
-            string directory = Path.GetDirectoryName(file);
-            if (directory != null)
-                Directory.CreateDirectory(directory);
 
             try {
-                using (var fileStream = File.Create(file)) {
+                using (var fileStream = CreateFileStream(file)) {
                     if (stream.CanSeek)
```

Comment:
I can't see what method this is but I don't think we should be seeking any stream. We also probably need a null check on FileStream

Reply:
I don't know why seeking the stream, but the original code did this. But what should it to do with null check, throw exception or just log message and do nothing.

Reply:
this should be fine because the last file create is outside of the try and it will blow up how it used to.. We should see what removing this seek does, we shouldn't ever seek a stream.

Reply:
Let's remove this seek.